### DISCUSSION TITLE
Phidget HUB0001, 2x isolated TC via 1100, & DAQ1301 Config

### DISCUSSION THREAD

**Discussion by zihyer** [artisan-roaster-scope/artisan#1336] [General]
Greetings from across the pond.

I'm stumped on an issue I caught running some test batches today on a seasoning blend.  After initially hooking everything up, it looked as though both ET and BT temps were working bc I was getting readings for both only a few degrees apart.  However, once I started to roast, it became evident that only the BT probe was changing while the ET remained flatlined:
![Screenshot 2023-12-07 182105](https://github.com/artisan-roaster-scope/artisan/assets/110182473/208de946-1a0c-46e6-a7f4-0d9d995921d0)

Here are my Phidget layout and connections:
![Screenshot 2023-12-07 181512](https://github.com/artisan-roaster-scope/artisan/assets/110182473/b80bd023-ca4b-453d-ba79-f5ec9e37a1f3)

I had originally configured to the closest machine setting in the presets to what I had going on which was the VINT TMP 1101 2x TC preset.  This gives me the two "U.U" readings so I started testing various device configs.  TMP1100 1x TC gave me two readings in the ambient range just a few 10ths of a degree apart so I **thought** I was good to go.  Shortly after firing the drum up, a problem became evident: 
![Screenshot 2023-12-07 182741](https://github.com/artisan-roaster-scope/artisan/assets/110182473/8f3a32ac-3881-4e4d-b72c-dfb719cb9c26)

I tried all the HUB device settings and tried tweaking settings under Device->Phidgets (tab) -> "1048... " and "1010..." panels for each but can't seem to get the ET temp to pull into the ET LCD field.  The closest I could come was this:
![Screenshot 2023-12-07 181607](https://github.com/artisan-roaster-scope/artisan/assets/110182473/ee1a5a39-af90-41ae-86b8-73821840ffdc)

And I did confirm that lower yellow reading is, in fact, the ET probe by spritzing some water on it and watching it quickly drop without affecting the BT.

Which I was able to do by adding an additional device and enabling the LCD for it:
![Screenshot 2023-12-07 181752](https://github.com/artisan-roaster-scope/artisan/assets/110182473/e0153612-1a6e-4846-8e21-a1eaeabbb1e5)

I've gone through the [Working with Phidgets](https://artisan-scope.org/devices/phidgets/) and [More Phidgets](https://artisan-roasterscope.blogspot.com/2017/12/more-phidgets.html) blogs and looked through the write-ups for the machine presets but I'm just not seeing how to get the ET reading to show up for the TMP1100 TC Phidget on the Hub port 1.

I have gone through trying all of the "Phidget HUB...", "Phidget IO... " and "Phidget TMP..." devices to no avail.  They all result in two "u.u" readings for BT and ET.  Setting directly to the "TMP 1100 1x TC" is the only setting that gives me anything "see above".

Can someone point me in the right direction here?  I feel like I'm super close and just haven't tried the right combination of settings between the ET/BT Meter and Phidgets tabs.

**Comment by poundy** [comment]
show us your ET/BT tab settings?

**Reply by zihyer** [reply]
Sure thing.  I've gone through all the others indicated above (and more) but his is the only configuration I have found that produces any readings other than "u.u / u.u": 
![Screenshot 2023-12-07 192459](https://github.com/artisan-roaster-scope/artisan/assets/110182473/e84e8891-9601-419f-b608-3125adaace6b).

And this produces a temp display similar to above (e.g. 75/185) however as the drum is cooling, the 185 (BT) drops accordingly but the ET is erroneous and seemingly unaffected by any real-world conditions as far as I can tell.  My guess that, as configured to read the 1100, the ET is some attempted read of a null channel.  Stated differently, it's trying to read channel 1 of a 01 of the TMP1100 however, since the 1100 is single channel, it defaults to a set value.

**Comment by MAKOMO** [comment]
You did not show us your "Symb BT/ET" tab. You need to link your ET data from "ETAddition" connected to Artisan extra device 2, channel 1 to the ET channel of the main device. Put `Y5` as ET symbolic formula "ET Y(x)" and check the HELP button for an explanation of symbolic variables as well as the corresponding blog post on [Symbolic Formulas: Basics, New Variables and Applications ](https://artisan-roasterscope.blogspot.com/2019/11/symbolic-formulas-basics-new-variables.html)

<img width="532" alt="Screenshot 2023-12-08 at 08 13 08" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/f5e9acdc-7d6a-4348-9149-a680a1a0bdd3">
